src/main/webapp/app/shared/pipes/sort-by.pipe.ts

```diff
@@ -65,11 +66,10 @@ export class SortByPipe implements PipeTransform {
     }
 
     /**
-     * Gets the duration of the exercise in minutes.
+     * Gets the duration of the submission in minutes.
      * @param exercise The exercise object.
      */
-    durationForExercise(exercise: any) {
-        // TODO: How does this work? An exercise does not have a completion date.
-        return this.momentDiff.transform(exercise.completionDate, exercise.participations[0].initializationDate, 'minutes');
+    durationForSubmission(submission: Submission) {
```

Comment:
I think this will break some other sorting. Can we make this method generic and passt `object: any` here? Then, we could distinguish between different types. I am pretty sure that duration is also displayed somewhere else.

Reply:
https://angular.io/guide/pipes#appendix-no-filterpipe-or-orderbypipe

I think we should think about removing this sortBy pipe anyway. Angular strictly says that it is a huge performance cost

Reply:
`The Angular team and many experienced Angular developers strongly recommend moving filtering and sorting logic into the component itself. The component can expose a filteredHeroes or sortedHeroes property and take control over when and how often to execute the supporting logic. Any capabilities that you would have put in a pipe and shared across the app can be written in a filtering/sorting service and injected into the component.`

Reply:
But maybe this should be done in a seperate ticket